src/learning/lda.rs

```diff
@@ -0,0 +1,250 @@
+//! Latent Diriclhet Allocation Module
+//!
+//! Contains an implementation of Latent Diriclhet Allocation (LDA) using
+//! [Gibbs Sampling](http://psiexp.ss.uci.edu/research/papers/sciencetopics.pdf)
+//!
+//! LDA is typically used for textual analysis.  It assumes that a topic can be modeled
+//! as a distribution of words, and that a document can be modeled as a distribution of
+//! topics.  Thus, the output is these distributions on the input documents.
+//!
+//! Gibbs sampling is a Morkov Chain Monto Carlo algorithm that iteratively approximates
+//! the above distributions.
+//!
+//! This module doesn't use any training.  It uses unsupervised learning to estimate
+//! the categories for a collection of documents.
+//!
+//! # Examples
+//!
+//! ```
+//! use rusty_machine::linalg::{Matrix, BaseMatrix, Vector};
+//! use rusty_machine::learning::UnSupModel;
+//! use rusty_machine::learning::lda::LDA;
+//!
+//! // Create a basic input array
+//! let input = Matrix::ones(5, 4);
+//!
+//! // Create a model that will find 5 topics, with parameters alpha=0.1, beta=0.1
+//! let lda = LDA::new(5, 0.1, 0.1);
+//!
+//! // No need to train the model (doing so is a no-op)
+//! lda.predict(&(input, 10)).unwrap();
+//! ```
+
+
+use linalg::{Matrix, Vector, BaseMatrix};
+use learning::{LearningResult, UnSupModel};
+use rulinalg::matrix::BaseMatrixMut;
+use rand::{Rng, thread_rng};
+
+/// Latent Dirichlet Allocation
+#[derive(Debug)]
+pub struct LDA {
+    topic_count: usize,
+    alpha: f64,
+    beta: f64,
+}
+
+/// An object which holds the results of Gibbs Sampling.
+///
+/// This object can then be used to get the distrbutions of
+/// topics over documents and words over topics
+#[derive(Debug)]
+pub struct LDAResult {
+    document_topic_count: Matrix<f64>,
+    topic_word_count: Matrix<f64>,
+    // The two vectors are simply used to reduce the amount of calculation that must be
+    // performed during the conditional probability step
+    topic_total_by_document: Vector<f64>,
+    word_total_by_topic: Vector<f64>,
+    alpha: f64,
+    beta: f64,
+}
+
+impl Default for LDA {
```

Comment:
As far as I can tell, there aren't any sensible constant defaults for LDA.  I only added this default implementation because the other models seemed to have it.  I could drop it if you like (or document it better, of course).